test/test_serializers/test_serializer_turtle.py

```diff
@@ -113,3 +113,24 @@ def test_turtle_namespace():
     assert "GENO:0000385" in output
     assert "SERIAL:0167-6423" in output
     assert r"EX:name_with_\(parenthesis\)" in output
+
+
+def test_issue948_escapeturtleprefixname():
+    foo = Namespace("http://www.example.com/foo/")
+    g = Graph()
+    g.bind("foo", foo)
+    g.add((foo["subj"], foo["pred"], foo["obj"]))
+    g.add((foo["subj"], foo["pred"], foo["obj/bar"]))
+    g.add((foo["subj"], foo["pred"], foo["obj#boo"]))
+    g1 = Graph()
+    g1.bind("foo", foo)
+    g1.parse(data=g.serialize(format="turtle"), format="turtle")
+    expected = r"""@prefix foo: <http://www.example.com/foo/> .
+
+foo:subj foo:pred foo:obj,
+        foo:obj\#boo,
+        foo:obj\/bar .
+
+"""
+    assert g.serialize(format="turtle") == expected
+    assert g1.serialize(format="turtle") == expected
```

Comment:
I was trying to add some more tests, and it seems this actually only works if there is more than 1 triple in the graph.

https://github.com/aucampia/rdflib/blob/4d31b1548340f513579354755e3c0bf5613d7f00/test/test_serializers/test_serializer_turtlelike.py#L18-L58

These tests fail if line 44 is removed.

Reply:
Looking into it further though.

Reply:
I'm fairly sure it is failing because the namespace is only in the list of checked namespaces the second time it is encountered. 

i.e. the namespace would only be added here:

https://github.com/RDFLib/rdflib/blob/1a60704c2ee6a3a1bf8a0824da5174037b87ed4e/rdflib/plugins/serializers/turtle.py#L319

I may be wrong though, but I'm generally not clear on why it is using a list of namespaces other than the list of namespaces provided by the namespace manager.